File: py/post.py
#!/usr/bin/env python3
'''Script for posting best result from state to site
'''
import sys
import requests

import state


BEATTHEHASH_URI = 'http://beatthehash.com/hash'

USAGE = '''post.py <username> <state_path>

    username     username to post to server with
    state_path   path to YAML state file
'''


def post_it(uri, username, state_path):
    '''Posts the value to the server

    Args:
        uri (str): uri of the server
        username (str): username to post with
        state_path (str): path to yaml state file
    '''
    last_state = state.State.load(state_path)

    data = {
        'username': username,
        'value': last_state.value
    }
    print(data)
    result = requests.post(uri, data=data)
    print(result.status_code)
    print(result.text)


def main():
    '''Perform post of best value from state to site
    '''
    if len(sys.argv) != 3:
        print(USAGE)
        exit(-1)

    username = sys.argv[1]
    state_path = sys.argv[2]

    post_it(BEATTHEHASH_URI, username, state_path)


if __name__ == '__main__':
    main()

File: go/src/mine/post/post.go
package main

import (
	"fmt"
	"io/ioutil"
	"mine/common"
	"mine/state"
	"net/http"
	"net/url"
	"os"
)

var usage = "post.py <username> <state_path>\n\n" +
	"username     username to post to server with\n" +
	"state_path   path to YAML state file\n"

var beatTheHashURI = "http://beatthehash.com/hash"

// postIt posts value to the server
func postIt(serverURL string, username string, value string) {
	data := url.Values{
		"username": {username},
		"value":    {value},
	}

	resp, err := http.PostForm(serverURL, data)
	common.Check(err)

	bodyBytes, err := ioutil.ReadAll(resp.Body)
	common.Check(err)

	fmt.Println(data)
	fmt.Println(resp.StatusCode)
	fmt.Println(string(bodyBytes))
}

func main() {
	args := os.Args
	if len(args) != 3 {
		fmt.Println(usage)
		os.Exit(-1)
	}

	username := os.Args[1]
	statePath := os.Args[2]

	lastState := state.State{}
	lastState.Load(statePath)
	postIt(beatTheHashURI, username, lastState.Value)
}
